### ISSUE TITLE
Certain byte values in Message.Body throw deserialization exception

### ISSUE THREAD

**Issue by seancolonello** [Azure/azure-functions-servicebus-extension#48]
In version 4.1.0 there are 77 byte values that when used in the body of a message, throw the following exception.

`System.Private.CoreLib: Exception while executing function: SendMail_In. Microsoft.Azure.WebJobs.Host: Exception binding parameter 'message'. Microsoft.Azure.WebJobs.ServiceBus: The Message with ContentType 'null' failed to deserialize to a string with the message: 'There was an error deserializing the object of type System.String. Unexpected end of file. Following elements are not closed: .'. System.Private.DataContractSerialization: There was an error deserializing the object of type System.String. Unexpected end of file. Following elements are not closed: . System.Private.DataContractSerialization: Unexpected end of file. Following elements are not closed: .`

Downgrading to version 4.0.0 fixes the problem. To reproduce I used the following code to push messages onto a Service Bus Queue (using the Microsoft.Azure.ServiceBus v4.1.1 library).

```c#
var client = new QueueClient(
    "Endpoint=[...]",
    "develop-sean");

// These are OK
for (int i = 0; i <= 127; i++)
    await client.SendAsync(new Message(new byte[] { (byte)i }));
for (int i = 194; i <= 244; i++)
    await client.SendAsync(new Message(new byte[] { (byte)i }));

// These throw the exception
for (int i = 128; i <= 193; i++)
    await client.SendAsync(new Message(new byte[] { (byte)i }));
for (int i = 245; i <= 255; i++)
    await client.SendAsync(new Message(new byte[] { (byte)i }));

await client.CloseAsync();
```

In essence, if I try to use a byte in the range of 128-193 or 245-255 the problem occurs. My receiving function app is .NET Core 3.1 with the following trigger.

```c#
[FunctionName(nameof(SendMail_In))]
public void SendMail_In(
    [ServiceBusTrigger("develop-sean", Connection = "Relia:ServiceBus:Connection")] Message message)
{
    //
}
```

**Comment by matthewhancock** [comment]
I'm having a similar issue which is closer to #52 which was closed as a dupe of this issue. I gzip my message before sending it to the queue. I'm not sure why it's even trying to parse the message body (which is a byte array) as a string within the process. Without additional context, I'm not sure [d95d2a3](https://github.com/Azure/azure-functions-servicebus-extension/commit/d95d2a3fa4baa98ea4a5c2ecc55cd3eca873ae8a) will resolve this.

` System.Private.CoreLib: Exception while executing function: [redacted]. Microsoft.Azure.WebJobs.Host: Exception binding parameter 'Message'. Microsoft.Azure.WebJobs.ServiceBus: The Message with ContentType 'application/gzip' failed to deserialize to a string with the message: 'There was an error deserializing the object of type System.String. The input source is not correctly formatted.'. System.Private.DataContractSerialization: There was an error deserializing the object of type System.String. The input source is not correctly formatted. System.Private.DataContractSerialization: The input source is not correctly formatted.`

**Comment by pseabury** [comment]
Same issue with deserialization as @matthewhancock - I compress my content with gzip and contsruct Messages from the resulting byte[].  Will c93ed8d fis this?  If so, when will that be available? 
 In the interim is downgrading the recommended workaround?  More details below.

I currently have a Producer generating a Microsoft.Azure.ServiceBus.Message [Assembly Microsoft.Azure.ServiceBus, Version=4.1.1.0, Culture=neutral, PublicKeyToken=7e34167dcc6d6d8c] and a Azure FunctionV3 ServiceBusTrigger consumer [Assembly Microsoft.Azure.WebJobs.ServiceBus, Version=4.1.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35].  Both are NETCore 3.1

The Message is created from a byte[] (gzipped json).  Before the consumer has a chance to process the message, the Host errors out trying to bind it to a Message object.

Producer
```
var msg = new Message(_serializer.Serialize(object).ToByteArray()) { MessageId = messageId, Label = myLabel };
```

Consumer
```
[Disable("ProcessPrimaryAdapterQueueFunc_Disable")]
[FunctionName("ProcessPrimaryAdapterQueueFunc")]
public async Task Run([ServiceBusTrigger("%AdapterQueueName%", Connection = PrimaryAdapterQueueConnection")]Message message, ILogger logger, ExecutionContext context)
{
     await ProcessQueueMessageImpl(message, logger, context);
}
```

Error

> [1/28/2020 2:42:15 PM] System.Private.CoreLib: Exception while executing function: ProcessSecondaryAdapterQueueFunc. Microsoft.Azure.WebJobs.Host: Exception binding parameter 'message'. Microsoft.Azure.WebJobs.ServiceBus: The Message with ContentType 'null' failed to deserialize to a string with the message: 'There was an error deserializing the object of type System.String. The input source is not correctly formatted.'. System.Private.DataContractSerialization: There was an error deserializing the object of type System.String. The input source is not correctly formatted. System.Private.DataContractSerialization: The input source is not correctly formatted.

**Comment by pseabury** [comment]
Thanks for the answers?

**Comment by TynThordin** [comment]
Same issue as @pseabury , will this be fixed?

**Comment by alrod** [comment]
The fix will be released in 4.1.2 next week. Workaround is using 4.0.0

**Comment by alrod** [comment]
4.1.1 released 3/3.
